### PR TITLE
feat: Configure platform in infracost comment command

### PR DESCRIPTION
**Pull request by mariusz89016**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -2,6 +2,12 @@ locals {
   #Repo attributes that are meant to simplify configuration rather than being actual repo options
   helper_options = ["allow_all_server_side_workflows", "terragrunt_atlantis_config", "infracost"]
 
+  infracost_parameters = {
+    github : { token_name : "github-token", environment_variable_name : "GITHUB_TOKEN" },
```
Comment (jakubigla) [inline main.tf:6]: Was thinking to to change the variable names to be more specific `INFRACOST_GITHUB_TOKEN` etc. `GITHUB_TOKEN` is too generic and if we want to be more granual with permissions, we could then pass token with permissions to only post comments.

**Review by jakubigla — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (mariusz89016) [reply main.tf:6]: yeah, you're right

(I was thinking about moving that `environment_variable_name` into variables (with default) but was worried about too many knobs there)

Comment (jakubigla) [reply main.tf:6]: yeah, very little value in exchange for possible technical debt

File: main.tf [hunk 2]
```diff
@@ -50,19 +56,28 @@ locals {
...
+                  "--repo $BASE_REPO_OWNER/$BASE_REPO_NAME",
+                  "--merge-request $PULL_NUM",
+                  "--path /tmp/$BASE_REPO_OWNER-$BASE_REPO_NAME-$PULL_NUM/'*'-infracost.json",
+                  format("--%s $%s",
+                    local.infracost_parameters[local._workflows[repo.workflow].infracost.platform].token_name,
+                    coalesce(
+                      local._workflows[repo.workflow].infracost.token_environment_variable,
+                      local.infracost_parameters[local._workflows[repo.workflow].infracost.platform].environment_variable_name)
+                  ),
+                  "--behavior new",
```
Comment (jakubigla) [inline main.tf:75]: can we add the `behaviour` to the variable object?

File: variables.tf [hunk 3]
```diff
@@ -136,7 +136,9 @@ variable "workflows" {
       file      = optional(string, "$SHOWFILE")
     }), {})
     infracost = optional(object({
-      enabled = optional(bool, false)
+      enabled                    = optional(bool, false)
+      platform                   = optional(string, "gitlab")
+      token_environment_variable = optional(string)
```
Comment (jakubigla) [inline variables.tf:141]: can we document this somehow? I don't think it would be clear to the new user, what is this used for.

**Review by jakubigla — COMMENTED** [review]
Minor comments. Other that that - I really like this PR

**Review by jakubigla — APPROVED** [review]

File: main.tf (hunk 2 above)
Comment (mariusz89016) [reply main.tf]: done

File: variables.tf (hunk 3 above)
Comment (mariusz89016) [reply variables.tf:141]: yeah, I've added readme - please check whether it's enough ☺️

**Pull request merged**